scripts/pull-gee-repository.sh

```diff
@@ -0,0 +1,370 @@
+#!/usr/bin/env bash
+
+#
+# Copyright 2017 Open GEE Contributors
+#
+# Licensed under the Apache license, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+#
+
+#===============================================================================
+#
+#          FILE:  pull-gee-repository.sh
+# 
+#         USAGE:  ./pull-gee-repository.sh 
+# 
+#   DESCRIPTION:  Acts as a frontend to allow for easier cloning of codebase 
+# 
+#  REQUIREMENTS:  Must not be root 
+#       VERSION:  1.0
+#===============================================================================
+
+
+#===============================================================================
+#       Constants  
+#===============================================================================
+THIS=$(basename "$0")
+USER_NAME=""
+PASSWORD=""
+DIR_NAME="earthenterprise"
+IS_SSH=1
+IS_HTTPS=2
+IS_MAIN=3
+IS_FORK=4
+SSH_DIR=".ssh"
+SSH_KEY_PRIV="id_rsa"
```

Comment:
While I'd believe that RSA is the dominant key type choice, it's not the only choice... I'd expect ECDSA to be a non-zero proportion of the user base?

Reply:
I suppose that the most appropriate validation would be to check that ~/.ssh is present and contains files?

Reply:
I don't see the benefit of checking what SSH keys a user has.  I think it's sufficient to use an SSH Git URL, and let `git` handle the rest.

Reply:
The specific keys aren't being checked at all; this section is really aimed at seeing *if* the user has keys. My comment really came down to that "id_rsa" is too specific.

Reply:
Thats fine, I just figured the standard git error message here is a bit cryptic, doesn't really explain much at first glance

Reply:
I know. And I don't see how checking if a user has keys gives us anything. It seems to be used only to replace the error message that `git` would give you, if you tried to clone over SSH, and Git couldn't find your keys.

Reply:
Fair enough. If you think thats sufficient than I can take it out